File: src/unit-test-data/Data.Tests/Tests.DataTest/Data/Test.Data.StructType.cs
#nullable enable

using Xunit;

namespace PrimeFuncPack.UnitTest.Data.Tests
{
    partial class TestDataTest
    {
        [Fact]
        public void SomeTextStructType_ExpectStructTypeTextIsSomeString()
        {
            var actual = TestData.SomeTextStructType;

            var expected = new StructType
            {
                Text = TestData.SomeString
            };

            Assert.Equal(expected, actual);
        }

        [Fact]
        public void LowerSomeTextStructType_ExpectStructTypeTextIsLowerSomeString()
        {
            var actual = TestData.LowerSomeTextStructType;

            var expected = new StructType
            {
                Text = TestData.LowerSomeString
            };

            Assert.Equal(expected, actual);
        }

        [Fact]
        public void AnotherTextStructType_ExpectStructTypeTextIsAnotherString()
        {
            var actual = TestData.AnotherTextStructType;

            var expected = new StructType
            {
                Text = TestData.AnotherString
            };

            Assert.Equal(expected, actual);
        }

        [Fact]
        public void UpperAnotherTextStructType_ExpectStructTypeTextIsUpperAnotherString()
        {
            var actual = TestData.UpperAnotherTextStructType;

            var expected = new StructType
            {
                Text = TestData.UpperAnotherString
            };

            Assert.Equal(expected, actual);
        }

        [Fact]
        public void NullTextStructType_ExpectDefaultStructTypeValue()
        {
            var actual = TestData.NullTextStructType;

            var expected = default(StructType);

            Assert.Equal(expected, actual);
        }
    }
}

File: src/unit-test-data/Data.FSharp.Tests/Tests.DataTest/Test.Data.StructType.fs
namespace PrimeFuncPack.UnitTest.Data.Tests.TestData

open Xunit
open PrimeFuncPack.UnitTest
open TestHelper

module DataStructTypeTest =

    [<Fact>]
    let ``TestData.SomeTextStructType should be with the text is the some string``() =
        let expected = StructType(Text = TestData.SomeString)
        TestData.SomeTextStructType |> shouldBeEqual expected

    [<Fact>]
    let ``TestData.LowerSomeTextStructType should be with the text is the lower some string``() =
        let expected = StructType(Text = TestData.LowerSomeString)
        TestData.LowerSomeTextStructType |> shouldBeEqual expected

    [<Fact>]
    let ``TestData.AnotherTextStructType should be with the text is the other string``() =
        let expected = StructType(Text = TestData.AnotherString)
        TestData.AnotherTextStructType |> shouldBeEqual expected

    [<Fact>]
    let ``TestData.UpperAnotherTextStructType should be with the text is the upper other string``() =
        let expected = StructType(Text = TestData.UpperAnotherString)
        TestData.UpperAnotherTextStructType |> shouldBeEqual expected

    [<Fact>]
    let ``TestData.NullTextStructType should be with the text is null``() =
        let expected = StructType()
        TestData.NullTextStructType |> shouldBeEqual expected

